addons/website_sale_collect/models/sale_order.py

```diff
@@ -61,6 +61,17 @@ def _check_cart_is_ready_to_be_paid(self):
             raise ValidationError(_("Some products are not available in the selected store."))
         return super()._check_cart_is_ready_to_be_paid()
 
+    def _get_cart_and_free_qty(self, product, line=None):
+        """ Override of `website_sale_stock` to get free_qty of the product from the warehouse that
+        was chosen rather than website's one.
+        """
```

Comment:
nitpicking
```suggestion
    def _get_cart_and_free_qty(self, product, line=None):
        """ Override of `website_sale_stock` to get free_qty of the product from the warehouse that
        was chosen rather than website's one.
        
        :param product.product product: The product
        :param sale.order.line line: The optional line
        """
```